src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Map/BrowseBuildingFloors/BrowseBuildingFloors.xaml.cs

```diff
@@ -0,0 +1,101 @@
+        description: "Display and browse through building floors from a floor-aware web map.",
+        instructions: "Use the spinner to browse different floor levels in the facility. Only the selected floor will be displayed.",
+        tags: new[] { "building", "facility", "floor", "floor-aware", "floors", "ground floor", "indoor", "level", "site", "story" })]
+    public partial class BrowseBuildingFloors
+    {
+        private const string _portalItem = "f133a698536f44c8884ad81f80b6cfc7";
+        private FloorManager _floorManager;
+
+        // Collection of floors.
+        private readonly Dictionary<string, FloorLevel> _floorOptions = new Dictionary<string, FloorLevel>();
+
+        public BrowseBuildingFloors()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            try
+            {
+                ArcGISPortal portal = await ArcGISPortal.CreateAsync();
+
+                // Get the portal item for a web map using its unique item id.
+                PortalItem mapItem = await PortalItem.CreateAsync(portal, _portalItem);
+                Map map = new Map(mapItem);
+
+                MyMapView.Map = map;
+
+                // Map needs to be loaded in order for floormanager to be used.
+                await MyMapView.Map.LoadAsync();
+                await MyMapView.Map.FloorManager.LoadAsync();
+
+                if (MyMapView.Map.FloorManager.LoadStatus == LoadStatus.Loaded)
+                {
+                    _floorManager = MyMapView.Map.FloorManager;
+
+                    // Use the dictionary to add the level's name as the key and the FloorLevel object with the associated level's name.
+                    foreach (FloorLevel level in _floorManager.Levels)
+                    {
+                        _floorOptions.Add(level.ShortName, level);
+                    }
+                }
+                // Provides an error message if the floor manager failed to load.
+                else if (MyMapView.Map.FloorManager.LoadStatus == LoadStatus.FailedToLoad)
+                {
+                    MessageBox.Show("Floor manager failed to load.", "Error", MessageBoxButton.OK, MessageBoxImage.Error);
+                    return;
+                }
+
+                FloorChooser.ItemsSource = _floorOptions.Keys;
+                FloorChooser.SelectedIndex = 0;
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
+            }
+        }
+
+        private void OnFloorChooserSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            // Get the name of the selected floor.
+            string selectedFloorName = e.AddedItems[0].ToString();
+
+            // Set all existing floors visibility to false.
+            foreach (FloorLevel level in _floorManager.Levels)
```

Comment:
It might make sense to be very clear about the design choice here. This code will hide all levels in all facilities in all sites before showing the selected level. In a lot of scenarios, it would make sense to keep all of the facilities in sync, so that if you have two buildings side-by-side, when you select level 2 in one, you see level 2 in all.